_posts/2025-08-01-classy-records.md

```diff
@@ -0,0 +1,53 @@
+---
+layout: post
+title: "Classy Records"
+---
+
+Record types are certainly one of my favourite newer C# features (they were actually released in 2020, in case you want to feel old once more). They were adopted eagerly for a good reason. However, I've seen enough times where a class would just have been the better choice due to the implementation. I just stumbled upon such a case when I found a record implemented something like this in a codebase:
+
+```csharp
+public record Addition(int X, int Y)
+{
+    private long? cachedResult = null;
+    public long Result => cachedResult ??= X + Y;
+}
+```
+
+Let's assume, that the computation of the addition would be really complicated and there is justifiable some reason to cache the result. Do you immediately spot the issue (knowing that there must be an issue may spoil that question...)?
+
+Let's write a test:
+
+```csharp
+[Test]
+public void AdditionTests()
+{
+    var addition1 = new Addition(3, 4);
+    Assert.That(addition1.Result, Is.EqualTo(7));
+}
+```
+
+This passing test just highlights how hard caching logic really would be to test properly in code that isn't just written for the sake of a blog post. Changing the test to the following brings up the problem:
```

Comment:
```suggestion
This passing test merely highlights how challenging it would be to test caching logic properly in code that isn't specifically written for the sake of a blog post. Changing the test to the following brings up the problem:
```